lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+DOCUMENTATION = '''
+---
+module: flatpak
+version_added: "2.4"
+requirements:
+    - flatpak
+author:
+    - "John Kwiatkoski (@jaykayy)"
+short_description: "Install and remove flatpaks."
+description:
+    - "The flatpak module allows users to manage installation and removal of flatpaks."
+options:
+  name:
+    description:
+      - "When state is set to `present`, `name` is best used as `http(s)` url format.
+        When set to `absent` the same `http(s)` will try to remove it using the
+        name of the flatpakref. However, there is no naming standard between
+        names of flatpakrefs and what the reverse DNS name of the installed flatpak
+        will be. Given that, it is best to use the http(s) url for name=`present`
+        and the reverse DNS name=`absent`. Alternatively reverse dns format can optimally
+        be used with `state=absent`
+        ex: name=org.gnome.gedit."
+    required: false
+    default: ''
+  remote:
+    description:
+      - "The flatpak remote repo to be used in the flatpak operation."
+    required: false
+    default: ''
+  state:
+    description:
+      -  "Set to `present` will install the flatpak and/or remote. 
+         Set to `absent` will remove the flatpak and/or remote."
+    required: false
+    default: present
+'''
+EXAMPLES = '''
+
+#install the spotify flatpak
+
+ - flatpak:
+    name:  https://s3.amazonaws.com/alexlarsson/spotify-repo/spotify.flatpakref
+    state: present
+
+#add the gnome remote andd install gedit flatpak
+ - flatpak:
+    name: https://git.gnome.org/browse/gnome-apps-nightly/plain/gedit.flatpakref
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+#remove the gedit flatpak and remote
+ - flatpak:
+    name: org.gnome.gedit
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+#remove the gedit package
+ - flatpak:
+    name: org.gnome.gedit
+    state: absent
+
+'''
+RETURN = '''
+reason:
+    description: on failure, the output for the failure
+    returned: failed
+    type: string
+    sample: "error while installing..."
+name:
+    description: "Name of flatpak given for the operation"
+    returned: always
+    type: string
+    sample: "https://git.gnome.org/.../gnome-apps/gedit.flatpakref"
+remote:
+    description: "remote of flatpak given for the operation"
+    returned: always
+    type: string
+    sample: "https://sdk.gnome.org/gnome-apps.flatpakrepo"
+'''
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def install_flat(flat):
+    command = "flatpak install -y --from {}".format(flat)
```

Comment:
Each of these functions creates a command but uses the relative executable `flatpak` but you have the absolute path below as `location` and you fail if it's not found. You could pass that location as an agrument or create a class wrapper around the functions. It would ensure that the initial flatpak path would always be used and possibly prevent environment issues.